### ISSUE TITLE
✨ [Feature] static file support use go:embed

### ISSUE THREAD

**Issue by devhaozi** [goravel/goravel#456]
### Before feedback (在反馈之前)

- [X] There are no features that I want to submit in Issues (当前 Issues 中没有我将要提交的新功能)

### Describe Feature (描述功能)

I want to use `go:embed` to embed some static files (e.g. translation files in the lang directory) directly into the binary at compile time, but the framework currently does not support this.

**Comment by jeff87218** [comment]
Here is my solution. For example, if you want to add go:embed for the public folder, you can add a file called assets.go (or another name) in the public folder.
![擷取](https://github.com/user-attachments/assets/6fccea7b-1503-49bf-8f31-0d77aa6bf015)
Then, add //go:embed * to a variable.
![擷取](https://github.com/user-attachments/assets/85b7efcd-d1a3-412f-9fcf-8874b65f4b2b)
Now you can use the static files in the router (or somewhere else).
![擷取](https://github.com/user-attachments/assets/85b6ebd7-3f20-44be-a823-7cea15a07e9c)

**Comment by devhaozi** [comment]
> Here is my temporary solution. For example, if you want to add go:embed for the public folder, you can add a file called assets.go (or another name) in the public folder. ![擷取](https://private-user-images.githubusercontent.com/29706585/355100695-6fccea7b-1503-49bf-8f31-0d77aa6bf015.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MTU3NzEsIm5iZiI6MTcyNjkxNTQ3MSwicGF0aCI6Ii8yOTcwNjU4NS8zNTUxMDA2OTUtNmZjY2VhN2ItMTUwMy00OWJmLThmMzEtMGQ3N2FhNmJmMDE1LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDEwNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2Zjc0ODYyYzMyZjZiM2FhNTUxNjc3ODY4MTA0MjZiMTMzZDkyZjdjN2RjODk1N2ZmMjIyYTJjMjAxMmM5ZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._Qo2MiHByoArsWCi4uSap9djTr2VZVNf-4luQGc5Fm4) Then, add //go:embed * to a variable. ![擷取](https://private-user-images.githubusercontent.com/29706585/355100902-85b7efcd-d1a3-412f-9fcf-8874b65f4b2b.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MTU3NzEsIm5iZiI6MTcyNjkxNTQ3MSwicGF0aCI6Ii8yOTcwNjU4NS8zNTUxMDA5MDItODViN2VmY2QtZDFhMy00MTJmLTlmY2YtODg3NGI2NWY0YjJiLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDEwNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NzM4YjI4NjE2ODAxNjM5YjI2MjllNGEwMDgwOWM1OWJhMDk5ZDJlMzJjOGFjYjhlNGRjOTQ0NWIyNDFhMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c6JOaGJYUw9nhvUJmktlbXIkqaV6SsXnQM6exXdn_ZA) Now you can use the static files in the router (or somewhere else). ![擷取](https://private-user-images.githubusercontent.com/29706585/355102707-85b6ebd7-3f20-44be-a823-7cea15a07e9c.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MTU3NzEsIm5iZiI6MTcyNjkxNTQ3MSwicGF0aCI6Ii8yOTcwNjU4NS8zNTUxMDI3MDctODViNmViZDctM2YyMC00NGJlLWE4MjMtN2NlYTE1YTA3ZTljLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDEwNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlYjFjZmRhYWMxMjQ5YWI2ZjMyMmM1MzdiMzJiYzdhMmUzOWJlN2NjYjcwYjRkZGFhYjEwNzc0MjIzMWZkZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.McSATJ23gOthKi7HQW2ROJ4XTNEjwP1Pd3vaw4yL4o8)
> 
> I hope there is a graceful way to do this.

Yes, but for some things (eg translation files) this is not supported.
